impl/client/src/main/javascript/web/pentaho/type/Context.js

```diff
@@ -953,6 +948,8 @@ define([
           if(config) type.constructor.implement(config);
 
           this._byTypeId[id] = InstCtor;
+          var alias = type.alias;
+          if(alias && !O.getOwn(this._byTypeId, alias)) this._byTypeId[alias] = InstCtor;
```

Comment:
My original intention would actually be better expressed if I had used `O_hasOwn` instead of `O.getOwn`. We should not need to check the value at all, just the existence of a key.